synfig-studio/src/gui/workarea.cpp

```diff
@@ -343,8 +343,17 @@ WorkArea::set_drag_mode(DragMode mode)
 {
 	if (drag_mode == mode) return;
 
+	// change cursor in drag window mode
+	if (mode == DRAG_WINDOW)
+		set_cursor(Gdk::FLEUR);
+
+	DragMode old_mode = drag_mode;
 	drag_mode = mode;
 
+	// reset cursor to previous cursor after finishing drag window mode
+	if (old_mode == DRAG_WINDOW)
+		set_cursor(previous_cursor);
```

Comment:
What about do not create this variable `old_mode`, and rename the parameter name to `new_drag_mode`?
```c++
if (new_drag_mode == DRAG_WINDOW)
	set_cursor(...);
else // reset cursor [...]
	set_cursor(...);
drag_mode = new_drag_mode;
```

Reply:
The problem here is that if `drag_mode == DRAG_WINDOW`, then `set_cursor` would have no effect. So I have first to change `drag_mode` then change the cursor.
Similarly, if `new_drag_mode == DRAG_WINDOW`, then I have to change the cursor first, then I have to change `drag_mode`. Which makes this awkward code order necessary.
Maybe I can define `set_cursor_forced` private function that would just change the cursor to make the code flow better.